src/helpers/ms-graph.js

```diff
@@ -16,13 +16,16 @@ const msGraphPlugin = {
 
     server.logger.info('Setting up ms-graph')
 
-    const agent = proxyAgent?.agent
-    const options = agent && {
-      proxyOptions: {
-        host: proxyAgent?.url.hostname,
-        port: proxyAgent?.url.port
-      }
-    }
+    const proxy = proxyAgent()
+    const agent = proxy?.agent
+    const options = agent
+      ? {
+          proxyOptions: {
+            host: agent.proxy.href,
+            port: agent.connectOpts.port
```

Comment:
Or `proxy.url.port`...

Reply:
so no port on `agent.proxy.port` - see above agent comment. This is only available on `agent.connectOpts.port`. I checked https://www.npmjs.com/package/https-proxy-agent and we are doing everything correct when instantiating this lib

Reply:
Now referencing the `proxy.url.href` 👍